test_runner/regress/test_timeline_archive.py

```diff
@@ -46,48 +48,46 @@ def test_timeline_archive(neon_simple_env: NeonEnv):
 
     assert exc.value.status_code == 404
 
-    # construct pair of branches to validate that pageserver prohibits
+    # construct a pair of branches to validate that pageserver prohibits
     # archival of ancestor timelines when they have non-archived child branches
-    parent_timeline_id = env.neon_cli.create_branch("test_ancestor_branch_archive_parent", "empty")
+    parent_timeline_id = env.neon_cli.create_branch("test_ancestor_branch_archive_parent")
 
     leaf_timeline_id = env.neon_cli.create_branch(
         "test_ancestor_branch_archive_branch1", "test_ancestor_branch_archive_parent"
     )
 
-    timeline_path = env.pageserver.timeline_dir(env.initial_tenant, parent_timeline_id)
-
     with pytest.raises(
         PageserverApiException,
         match="Cannot archive timeline which has non-archived child timelines",
     ) as exc:
-        assert timeline_path.exists()
-
         ps_http.timeline_archival_config(
             tenant_id=env.initial_tenant,
             timeline_id=parent_timeline_id,
             state=TimelineArchivalState.ARCHIVED,
         )
 
-    assert exc.value.status_code == 412
+    assert exc.value.status_code == 412 if unsharded else 409
```

Comment:
does this make any sense to have different behaviour?

Reply:
Yes, because the storage controller returns 409 in the case that there is errors coming from the pageservers.

We should only return 412 in the storage controller if we ensure that all shards return the same error, which is something you requested to be removed in the comments.

Reply:
alright I've removed this in favour of matching on the API error.

Reply:
nice.